Document components for bundle optimization (#10569)
chartjs/Chart.js: docs/getting-started/integration.md

> Chart.js 3 is tree-shakeable, so it is necessary to import and register the controllers, elements, scales and plugins you are going to use.

Reviewer: is it weird that this refers to Chart.js 3? At the very least I think it should say "Chart.js 3+", but maybe we should change it to just "Chart.js"?

_Originally posted by @benmccann in https://github.com/chartjs/Chart.js/pull/9721#discussion_r938841542_

### Quick start

If you don't care about the bundle size, you can use the `auto` package ensuring all features are available:

```javascript
import Chart from 'chart.js/auto';
```

> Or if you are planning to optimize the bundle later, you can start by importing and registering `registerables`, which also ensures all features are available:
> 
> ```javascript
> import { Chart, registerables } from 'chart.js';
> Chart.register(...registerables);
> ```
> 

Reviewer 2: I feel like it just makes it more complicated by documenting more ways to do the same thing. We should either document this or auto, but not both. I'd be fine dropping either
```suggestion
```

### Bundle optimization

> When optimizing the bundle, instead of the `registerables`, you need to import and register the components that are needed in your application.

Reviewer 2: if we remove `registerables` above, we should remove mention of it here as well
```suggestion
When optimizing the bundle, you need to import and register the components that are needed in your application.
```

> The options are categorised (inspect `registerables`) into controllers, elements, plugins, scales. You can pick and choose many of these, e.g. if you are not going to use tooltips, don't import and register the `Tooltip` plugin. But each type of chart has its own bare-minimum requirements (typically the type's controller, element(s) used by that controller and scale(s)):

Reviewer 2: People shouldn't have to inspect our code. We should just document it
```suggestion
The options are categorized into controllers, elements, plugins, scales. You can pick and choose many of these, e.g. if you are not going to use tooltips, don't import and register the `Tooltip` plugin. But each type of chart has its own bare-minimum requirements (typically the type's controller, element(s) used by that controller and scale(s)):
```

Author: They'd be inspecting the types, which could be considered part of documentation. But I'm ok either way.

Fyi, it will be about 2 weeks before I'm at a computer.

* Bar chart
  * `BarController`
  * `BarElement`

>   * Default scales: x: `CategoryScale`, y: `LinearScale`

Reviewer 2: it's a little weird to have multiple `:`. I would change one to a `-`. (and below as well)
```suggestion
  * Default scales - x: `CategoryScale`, y: `LinearScale`
```

* Bubble chart
  * `BubbleController`
  * `PointElement`
  * Default scales: x/y: `LinearScale`
* Doughnut chart
  * `DoughnutController`
  * `ArcElement`
  * Not using scales
* Line chart
  * `LineController`
  * `LineElement`
  * `PointElement`
  * Default scales: x: `CategoryScale`, y: `LinearScale`
* Pie chart
  * `PieController`
  * `ArcElement`
  * Not using scales
* PolarArea chart
  * `PolarAreaController`
  * `ArcElement`
  * Default scale: r: `RadialLinearScale`
* Radar chart
  * `RadarController`
  * `LineElement`
  * `PointElement`
  * Default scale: r: `RadialLinearScale`
* Scatter chart
  * `ScatterController`
  * `PointElement`
  * Default scales: x/y: `LinearScale`

> The plugin imports are:

Reviewer 2: are some of these required and some optional?
```suggestion
The available plugins are:
```

Reviewer: All plugin options are optional
